code/controllers/subsystem/minor_mapping.dm

```diff
@@ -16,26 +16,20 @@ SUBSYSTEM_DEF(minor_mapping)
 /datum/controller/subsystem/minor_mapping/proc/trigger_migration(num_mice=10)
 	var/list/exposed_wires = find_exposed_wires()
 
-	var/mob/living/basic/mouse/mouse
 	var/turf/open/proposed_turf
-
-
 	while((num_mice > 0) && exposed_wires.len)
 		proposed_turf = pick_n_take(exposed_wires)
-
-		if(!istype(proposed_turf))
+		if (!istype(proposed_turf) || \
+			!proposed_turf.air.has_gas(/datum/gas/oxygen, 5) || \
+			proposed_turf.temperature > NPC_DEFAULT_MAX_TEMP || \
+			proposed_turf.temperature < NPC_DEFAULT_MIN_TEMP)
```

Comment:
this is ugly to me (wack indentation for something like this IMO), just spin it out into it's own proc that returns TRUE/FALSE depending on turf validity like

```txt
ssminormapping/proc/valid_turf(turf/open/checkable)
    if(!istype(checkable)
        return FALSE
    var/turf_gasmix = checkable.air // (this one is optional actually)
    var/temperature = checkable.temperature // (this one makes good logical sense we save 0.00001 CPU cycles with this cache
     if(...)
```

don't have to do it exactly the way i did it but this is much cleaner to me footprint-wise.